opendevin/core/config.py

```diff
@@ -297,9 +336,21 @@ def set_attr_from_env(sub_config: Any, prefix=''):
                 nested_sub_config = getattr(sub_config, field_name)
 
                 # the agent field: the env var for agent.name is just 'AGENT'
-                if field_name == 'agent' and 'AGENT' in env_or_toml_dict:
+                if field_name == 'agent' and env_var_name in env_or_toml_dict:
                     setattr(nested_sub_config, 'name', env_or_toml_dict[env_var_name])
 
+                old_configs = ['INITIALIZE_PLUGINS']
+                if field_name == 'sandbox':
+                    for old_config in old_configs:
+                        if (
+                            old_config.lower() in nested_sub_config.__annotations__
+                            and old_config in env_or_toml_dict
+                        ):
+                            suggested_name = f'{field_name}_{old_config}'.upper()
+                            logger.error(
+                                f'Please migrate {old_config} config to {suggested_name} = {env_or_toml_dict[old_config]}'
+                            )
+                            exit(1)
```

Comment:
If I see this right, this code applies to `_initialize_plugins_` from env (uppercase), and not from toml (lowercase)? Couldn't it be in the toml?

Separately, personally, I am not sure that we need to give the user an error over it... Why not postpone it if you don't think it should be supported now?

Reply:
I have commented out for a separate PR.